src/beacon.cpp

```diff
@@ -61,3 +61,12 @@ std::string GetStoredBeaconPublicKey(const std::string& cpid)
 {
     return GetArgument("publickey" + cpid + GetNetSuffix(), "");
 }
+
+void UpdateBeaconKeysInMemory(
+        const std::string &cpid,
+        const std::string &pubKey,
+        const std::string &privKey)
+{
+    mapArgs["-publickey" + cpid + GetNetSuffix()] = pubKey;
```

Comment:
Alright, this is a bit messy. The public key is retrieved from cache while the private key 

For the public key I think you want to use [WriteCache](https://github.com/gridcoin/Gridcoin-Research/blob/master/src/main.cpp#L8599) from main.cpp with the following arguments:
- section: "beacon"
- key: cpid (no testnet)
- value: public key
- locktime: `GetTime()`

For the private key you probably want to do like you did but through a new `SetArgument` function in main.cpp: 
 `SetArgument("publickey" + cpid, pubKey)` (no net suffix). Do it for both the public- and private key since they are both loaded from config for now. I'm not sure why the public key is stored in config.